File: src/main/java/racingcar/domain/Name.java
package racingcar.domain;

public record Name(String name) {
    private static final int MAX_NAME_LENGTH = 5;

    public Name {
        validateName(name);
    }

    private void validateName(String name) {
        if (name == null || name.isBlank()) {
            throw new IllegalArgumentException("이름은 빈 문자열이 될 수 없습니다.");
        }

        if (name.length() > MAX_NAME_LENGTH) {
            throw new IllegalArgumentException("이름은 " + MAX_NAME_LENGTH + "자 이하여야 합니다.");
        }
    }
}

File: src/main/java/racingcar/domain/Name.kt
package racingcar.domain

data class Name(
    val value: String
) {
    init {
        require(value.isNotBlank()) { "이름은 빈 값일 수 없습니다." }
        require(value.length in MINIMUM_LENGTH..MAXIMIN_LENGTH) { "이름의 길이는 1 ~ 5 사이여야 합니다." }
    }

    companion object {
        private const val MINIMUM_LENGTH = 1
        private const val MAXIMIN_LENGTH = 5
    }
}
